app_feup/lib/view/Pages/LoginPageView.dart

```diff
@@ -26,46 +30,68 @@ class LoginPageView extends StatelessWidget {
-          child: Padding(
-              padding: EdgeInsets.only(
-                  left: queryData.size.width / 8,
-                  right: queryData.size.width / 8,
-                  top: queryData.size.height / 6,
-                  bottom: queryData.size.height / 6),
-              child: Flex(
-                direction: Axis.vertical,
-                children: <Widget>[
-                  createTitle(queryData, context),
-                  Spacer(),
-                  Form(
-                    key: this.formKey,
-                    child: Column(children: [
-                      createUsernameInput(context),
-                      Padding(
-                          padding: EdgeInsets.only(
-                              bottom: queryData.size.height / 35)),
-                      createPasswordInput(),
-                    ]),
-                  ),
-                  Padding(
-                      padding:
-                          EdgeInsets.only(bottom: queryData.size.height / 35)),
-                  createSaveDataCheckBox(),
-                  Spacer(),
-                  createLogInButton(queryData),
-                  Spacer(),
-                  createStatusWidget(context)
-                ],
-              )),
-        ));
+    return new Scaffold(
+      backgroundColor: primaryColor,
+      body: WillPopScope(
+        child: Padding(
+          padding: EdgeInsets.only(
+            left: queryData.size.width / 8,
+            right: queryData.size.width / 8),
+          child: new ListView(
+            children: getWidgets(context, queryData),
+          )
+        ), onWillPop: () => onWillPop(context))
+
+    );
+  }
+
+  List<Widget> getWidgets(BuildContext context, MediaQueryData queryData){
+    List<Widget> widgets = new List();
+
+    widgets.add(Padding(padding: EdgeInsets.only(bottom: queryData.size.height/20)));
+    widgets.add(createTitle(queryData, context));
+    widgets.add(Padding(padding: EdgeInsets.only(bottom: queryData.size.height/35)));
+    widgets.add(getLoginForm(queryData, context));
+    widgets.add(Padding(padding: EdgeInsets.only(bottom: queryData.size.height/35)));
+    widgets.add(createSaveDataCheckBox());
+    widgets.add(Padding(padding: EdgeInsets.only(bottom: queryData.size.height/15)));
+    widgets.add(createLogInButton(queryData));
+    widgets.add(Padding(padding: EdgeInsets.only(bottom: queryData.size.height/35)));
+    widgets.add(createStatusWidget(context));
+    widgets.add(Padding(padding: EdgeInsets.only(bottom: queryData.size.height/10)));
+
+    return widgets;
+  }
+
+  void displayToastMessage(BuildContext context, String msg) {
+    Toast.show(
+      msg,
+      context,
+      duration: Toast.LENGTH_LONG,
+      gravity: Toast.BOTTOM,
+      backgroundColor: toastColor,
+      backgroundRadius: 16.0,
+      textColor: Colors.white,
+    );
+  }
+
+  Future<void> exitAppWaiter() async{
+    exitApp = true;
+    await new Future.delayed(Duration(seconds: 2));
```

Comment:
Since the toast is dependent on the current constant length_long, use this variable here as the duration of the future. Even though it has the same value, it is better to avoid any unexpected behaviour in the future